homeassistant/components/worldclock/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow for World clock."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any, cast
+import zoneinfo
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME, CONF_TIME_ZONE
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaCommonFlowHandler,
+    SchemaConfigFlowHandler,
+    SchemaFlowFormStep,
+)
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+)
+
+from .const import CONF_TIME_FORMAT, DEFAULT_NAME, DEFAULT_TIME_STR_FORMAT, DOMAIN
+
+
+async def validate_duplicate(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate already existing entry."""
+    handler.parent_handler._async_abort_entries_match({**handler.options, **user_input})  # noqa: SLF001
+
+    return user_input
+
+
+async def get_schema(handler: SchemaCommonFlowHandler) -> vol.Schema:
+    """Get available timezones."""
+    get_timezones: list[str] = list(
+        await handler.parent_handler.hass.async_add_executor_job(
+            zoneinfo.available_timezones
+        )
+    )
+    return vol.Schema(
+        {
+            vol.Required(CONF_NAME, default=DEFAULT_NAME): TextSelector(),
+            vol.Required(CONF_TIME_ZONE): SelectSelector(
+                SelectSelectorConfig(
+                    options=get_timezones, mode=SelectSelectorMode.DROPDOWN, sort=True
+                )
+            ),
+        }
+    ).extend(DATA_SCHEMA_OPTIONS.schema)
+
+
+DATA_SCHEMA_OPTIONS = vol.Schema(
+    {vol.Optional(CONF_TIME_FORMAT, default=DEFAULT_TIME_STR_FORMAT): TextSelector()}
```

Comment:
Should we make this a selector with some preset to pick from?

(As in key=format, value=example) and allow for custom?

As in, this is pretty technical, I'm just trying to think if we can make it simpler.

Reply:
Second thought: This should maybe be removed? As make it an actual timestamp sensor?

Reply:
That doesn't work as then it would convert it to local time hence the reason it's a string value. 
But I like the idea to provide some defaults for making it easier for the user than coming up just empty.